### PR TITLE
Remove obsolete TOML code

### PR DESCRIPTION
**Pull request by dime10**
This PR aims to remove obsolete TOML related code and also resolve incompatibilities with the latest lightning device versions.

- The verifier is relaxed to only test agreement of basic gates (no adjoint/control) between the toml and python device spec.
 - This is needed to resolve the overlapping gate issue introduced by https://github.com/PennyLaneAI/pennylane-lightning/pull/743
 - and also allows removing the patch for Lightning Kokkos that exists solely to satisfy the verifier.

- The obsolete toml files in the catalyst repo for lightning devices is also removed.

- MCMs are marked as un-invertible and un-controllable, since they are non-unitary.

Requires: https://github.com/PennyLaneAI/pennylane-lightning/pull/758

[sc-65114]

### CODE REVIEW

File: frontend/catalyst/device/qjit_device.py [hunk 1]
```diff
@@ -443,50 +443,29 @@ def execute(self, circuits, execution_config):
...
 
     Returns:
-        List: A list of strings with names of supported operations with Adjoint and C gates
-        removed.
+        Set: filtered set of operation names
     """
+    control = re.compile(r"^C\(.*\)$")
     adjoint = re.compile(r"^Adjoint\(.*\)$")
     c_adjoint = re.compile(r"^C\(Adjoint\(.*\)\)$")
     adjoint_c = re.compile(r"^Adjoint\(C\(.*\)\)$")
```
Comment (dime10) [inline frontend/catalyst/device/qjit_device.py:458]: Redundant:
```suggestion
```

**Comment by dime10** [comment]
The current development version of Lightning-Kokkos does not pass the runtime tests:
```
catalyst/runtime/tests/Test_LightningMeasures.cpp:1386: FAILED: due to unexpected exception with message:
  [catalyst/runtime/lib/backend/lightning/lightning_kokkos/LightningKokkosSimulator.cpp][Line:303][Function:PartialProbs]
  Error in Catalyst Runtime: Invalid size for the pre-allocated partial-probabilities
```

File: frontend/catalyst/device/qjit_device.py [hunk 2]
```diff
@@ -443,50 +443,29 @@ def execute(self, circuits, execution_config):
...
-    len_of_sets = [len(arg) for arg in args]
-    if sum(len_of_sets) == len(union):
-        return
-
-    overlaps = set()
-    for s in set_of_sets:
-        overlaps.update(s - union)
-        union = union - s
+    def is_not_modifier(op):
+        return not (
```
Comment (grwlf) [inline frontend/catalyst/device/qjit_device.py:461]: Should we match a single regexpr here?  Something like `\^(C(.*)|Adjoint(.*)$`

Comment (dime10) [reply frontend/catalyst/device/qjit_device.py:461]: Sure 👍

**Review by lillian542 — APPROVED** [review]
🎉

**Pull request merged**